### PR TITLE
Feature/stop jndi search timer if trigger mode and inbound started

### PR DESCRIPTION
**Pull request by aliothe**
If inbound mode is trigger and inbound has started, there is no need for the JNDI search timer to continue running.
This since when running in trigger mode, no dynamic deployments are expected.

### CODE REVIEW

File: casual/casual-api/src/main/java/se/laz/casual/config/Startup.java [hunk 1]
```diff
@@ -101,6 +103,7 @@ public Builder withServices( List<String> services )
 
         public Startup build()
         {
+            mode = (null == mode) ? Mode.fromName(Optional.ofNullable(System.getenv(CASUAL_INBOUND_STARTUP_MODE_ENV_NAME)).orElseGet(Mode.IMMEDIATE::getName)) : mode;
```
Comment (TobiasLeo) [inline casual/casual-api/src/main/java/se/laz/casual/config/Startup.java]: We already have ConfigurationService that can and should serve this config. I can see though that the code calling this from Inbound is some "older" code not touched by this PR. Also, wouldn't this fail to read the inbound startup mode if it is configured by file?

File: casual/casual-inbound-handler-casual-service/build.gradle [hunk 2]
```diff
@@ -19,9 +19,11 @@ dependencies {
   implementation project(":casual:casual-inbound-handler-api")
   implementation project(":casual:casual-service-discovery-extension")
   compileOnly libs.javaee_api
-
+  compileOnly libs.gson
+    
   testImplementation libs.javaee_api
   testImplementation libs.gson
+  testImplementation libs.system_lambda
```
Comment (TobiasLeo) [inline casual/casual-inbound-handler-casual-service/build.gradle:26]: Is system lambda actually ever used? I can't see it in any tests in this version, and seems to work fine locally even if I remove it

File: casual/casual-api/src/main/java/se/laz/casual/config/Startup.java (hunk 1 above)
Comment (aliothe) [reply casual/casual-api/src/main/java/se/laz/casual/config/Startup.java]: Fixed, going with config files and also testing using config files and overriding the inbound startup mode via env var.

File: casual/casual-inbound-handler-casual-service/build.gradle (hunk 2 above)
Comment (aliothe) [reply casual/casual-inbound-handler-casual-service/build.gradle:26]: It was earlier and now it is again :)

**Review by TobiasLeo — APPROVED** [review]

**Pull request merged**